core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -158,6 +163,32 @@ public KafkaSchemaRegistry(SchemaRegistryConfig config,
     this.masterNodeSensor.add(m, new Gauge());
   }
 
+  /**
+   * A Schema Registry instance's identity is in part the port it listens on. Currently the
+   * port can either be configured via the deprecated port configuration, or via the listeners
+   * configuration.
+   *
+   * This method checks if listeners is configured, and if it is, uses the port from the first listener.
+   * If listeners isn't configured, the deprecated port configuration is used.
+   *
+   * In theory, any port from any listener would be sufficient. Choosing the first, instead of say the last,
+   * is arbitrary.
+   */
+  public static int getPortForIdentity(int port, List<String> listeners) {
+    // TODO: once RestConfig.PORT_CONFIG is deprecated, remove the port parameter.
+    if (listeners.isEmpty()) {
+      return port;
+    } else {
+      URI uri;
+      try {
+        uri = new URI(listeners.get(0));
+      } catch (URISyntaxException use) {
+        throw new ConfigException("First listener could not be parsed into a URI: " + listeners.get(0));
+      }
+      return uri.getPort();
```

Comment:
What if the port isn't defined in the URI? This method will return `-1` but that presumably isn't the value we want to use? Or do we require that `listeners` includes ports?

Or, as an alternative to this code, `rest-utils`' `Application` has a `parseListeners` that handles other checks. Should we just use the output of that method and centralize all this work there? Seems better than having to duplicate it across projects (e.g. will kafka-rest end up with similar problems?).

This also makes me wonder if we handle default ports properly in Kafka with listeners if the listener URL doesn't specify it explicitly, or if we even have an idea of what the expected behavior should be in that case.